testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -11692,9 +11692,9 @@ void assertSubFunReturns(boolean binary, String s, int start,
       f12.checkScalar("NVL2(NULL, 3.0, 2.111)", "2.111", "DECIMAL(4, 3) NOT NULL");
       f12.checkScalar("NVL2(3.111, 3.1415926, 2.111)", "3.1415926", "DECIMAL(8, 7) NOT NULL");
 
-      f12.checkNull("nvl2('ab', CAST(NULL AS VARCHAR(6)), 'def')");
-      f12.checkNull("nvl2(NULL, 'abc', NULL)");
-      f12.checkNull("nvl2(NULL, NULL, NULL)");
+      f12.checkNullValueWithNullableType("nvl2('ab', CAST(NULL AS VARCHAR(6)), 'def')");
```

Comment:
so you will fix this in another PR?

Reply:
There are two ways to fix this issue. 
One way is to first fix all function issues with `checkNullValueWithNullableType`, and then strengthen the 'checkNull' method to detect return data type is nullable, then delete  `checkNullValueWithNullableType`.
The other way is to directly delete the 'checkNullValueWithNullableType' method, and then use different methods to retest `checkNull`, such as `check`, which will retest the value and return type. The `checkNull` method is meaningless.
Do you have any suggestions?

Reply:
Maybe it's simplest to just add a type check to checkNull.

Reply:
Yes. I will add a type check to `checkNull` after deleting `checkNullValueWithNullableType`.Not now, because if we add type check in `checkNull` now, The unit test will not pass.